gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -886,6 +892,21 @@ public JsonWriter newJsonWriter(Writer writer) throws IOException {
     if (prettyPrinting) {
       jsonWriter.setIndent("  ");
     }
+    switch (newlineStyle) {
+      case CURRENT_OS:
+        jsonWriter.setNewline(System.lineSeparator());
+        break;
+      case WINDOWS:
+        jsonWriter.setNewline("\r\n");
+        break;
+      case OLD_MACOS:
+        jsonWriter.setNewline("\r");
+        break;
+      case MACOS_AND_LINUX: // intentional fallthrough
+      default:
```

Comment:
If this would be a new library then the best `default` would probably be the same as `CURRENT_OS`.
But changing this after 14 years would probably break too many things.

For what ECMAScript says about line terminators see
https://262.ecma-international.org/5.1/#sec-7.3
There is nothing there giving special status to `LF` (`U+000A`)

Reply:
Just my personal opinion, but I am rather skeptical towards OS defaults. They have caused a lot trouble for the Java ecosystem in the past, and are still causing trouble (e.g. `String.toUpperCase()` or default encoding). If users want to have OS defaults, then they should explicitly specify that.

So I think `LF` is a good default.

Reply:
Ack.

We might disagree on the OS defaults.
But since we both end up with the same conclusion (using different arguments), it's all good :-)

Thanks,
M.